src/mtrie.c

```diff
@@ -118,47 +118,56 @@ char *return_matches_m(trie_t *unused_t, char *key)
 // Text: and there was an ant; searches for 'an'
 // GList: and->an->ant
 // rank according to relative position in text or alphabetically?
-GList *return_matches(trie_t *trie, char *key)
+list_t return_matches(trie_t *trie, char *key)
 {
-	// just to silence warning
-	trie_t *temp = trie;
-	trie = temp;
+    // just to silence warning
+    trie_t *temp = trie;
+    trie = temp;
 
-    GList *result = NULL;
+    list_t result;
+    list_init(&result);
     // a->an->at->and
     if (strcmp(key, "a") == 0) {
```

Comment:
Every use of `strcmp` in this file needs to be changed to `strncmp`.

Reply:
done